x-pack/metricbeat/module/googlecloud/stackdriver/response_parser.go

```diff
@@ -0,0 +1,104 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package stackdriver
+
+import (
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/golang/protobuf/ptypes"
+	"google.golang.org/genproto/googleapis/monitoring/v3"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+func newIncomingFieldExtractor(l *logp.Logger) *incomingFieldExtractor {
+	return &incomingFieldExtractor{logger: l}
+}
+
+type incomingFieldExtractor struct {
+	logger *logp.Logger
+}
+
+type KeyValuePoint struct {
+	Key       string
+	Value     interface{}
+	Labels    common.MapStr
+	ECS       common.MapStr
+	Timestamp time.Time
+}
+
+// extractTimeSeriesMetricValues valuable to send to Elasticsearch. This includes, for example, metric values, labels and timestamps
+func (e *incomingFieldExtractor) extractTimeSeriesMetricValues(resp *monitoring.TimeSeries) (points []KeyValuePoint, err error) {
+	points = make([]KeyValuePoint, 0)
+
+	for _, point := range resp.Points {
+		// Don't add point intervals that can't be "stated" at some timestamp.
+		ts, err := e.getTimestamp(point)
+		if err != nil {
+			e.logger.Warn(err)
```

Comment:
I purposedly didn't add more context because I thought that the error is clear enough: `error trying to parse timestamp '%#v' from metric` The only possible context to this message is to add (when logging) "error getting timestamp" which is redundant.

A metric without timestamp is purposedly discarded. Any "guessing" attempt is just a mistake that can lead to unexpected results to the users.